// @sheet-importer/lib/src/components/Modal.tsx
import React from 'react'

interface ConfirmModalProps {
  isOpen: boolean
  title?: string
  message: string
  onCancel: () => void
  onConfirm: () => void
}

export const Modal: React.FC<ConfirmModalProps> = ({
  isOpen,
  title = 'Confirm',
  message,
  onCancel,
  onConfirm,
}) => {
  if (!isOpen) return null

  return (
    <div className="fixed inset-0 bg-black bg-opacity-50 flex items-center justify-center z-50">
      <div className="bg-white p-6 rounded-md shadow-md w-full max-w-md">
        <h2 className="text-lg font-semibold mb-4">{title}</h2>
        <p className="mb-6">{message}</p>
        <div className="flex justify-end gap-4">
          <button
            onClick={onCancel}
            className="px-4 py-2 text-gray-700 border border-gray-300 rounded hover:bg-gray-100"
          >
            Cancel
          </button>
          <button
            onClick={onConfirm}
            className="px-4 py-2 bg-blue-600 text-white rounded hover:bg-blue-700"
          >
            Confirm
          </button>
        </div>
      </div>
    </div>
  )
}
